internal/config/instance.go

```diff
@@ -113,13 +113,20 @@ func (i *Instance) GetThenSet(key string, valueF func(currentValue interface{})
 
 const CancelSet = "__CANCEL__"
 
-func (i *Instance) setWithCallback(key string, valueF func(currentValue interface{}) (interface{}, error)) error {
+func (i *Instance) setWithCallback(key string, valueF func(currentValue interface{}) (interface{}, error)) (rerr error) {
+	func() {
+		if rerr != nil {
```

Comment:
Isn't this always going to be `false` at this point in the function? Do you want to `defer` the anonymous function instead of calling it here?

Reply:
Woops, that was indeed meant to be a defer. Good eye!